kotlinx-coroutines-core/jvm/src/debug/ArtificialStackFrames.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.debug
+import kotlinx.coroutines.*
+
+/**
+ * A collection of artificial stack trace elements to be included in stack traces by the coroutines machinery.
+ */
+@InternalCoroutinesApi
+internal class ArtificialStackFrames {
```

Comment:
`ArtificialStackFrames` is indeed not very precise name.

I'd suggest preparing a small list of potential names and discuss it for 15 minutes during our regular DM.
Or it can even be done asynchronously in Slack.
A few potential names: `CoroutinesDebugging`, `DebugStackTraces` and `CoroutinesKnit`.

Another solution could be to somehow highlight these stackframes visually in user stack traces, e.g. by getting rid of package FQN. It won't clutter IDEA experience, because this class is internal anyway, but will provide a clear visual separator:
![image](https://user-images.githubusercontent.com/2780116/103210941-0e20b780-4918-11eb-96ab-91e35d9240e5.png)